tools/migration/sender-jid-from-mam-message.escript

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env escript
+
+-mode(compile).
+-compile(nowarn_deprecated_function).
+
+-include_lib("xmerl/include/xmerl.hrl").
+
+main([Type]) ->
+    case os:getenv("DEBUG") of
+        "1" ->
+            {ok, DebugFile} = file:open("/tmp/script-debug", [write, append]),
+            put(debug_file, DebugFile);
+        _ ->
+            ok
+    end,
+    try
+        ok = io:setopts([binary]),
+        loop(Type)
+    catch
+        C:R ->
+            debug(C, R, erlang:get_stacktrace(), #{ type => unrecoverable_error })
+    after
+        file:close(get(debug_file))
```

Comment:
will you ever enter this part of code, as (at least form instruction) you need to interrupt/kill process the script to stop it?

Reply:
It will enter this part when the processing loop crashes for some unexpected reason. Proper file close ensures that output buffer will be flushed and the reason will be logged for further debugging.